src/map/Map.cpp

```diff
@@ -0,0 +1,210 @@
+/*------------------------------------------------------------------------------
+* Source: Map.cpp
+*
+* Functions:
+*       Map(string file);
+*       int loadFileData();
+*       void genWalls(const struct MapPoint wstart[MAX_WALLS]);
+*       string getFile();
+*       void setFile(string f);
+*       void mapLoadToGame();
+*       void printData();
+*
+* Date: March 23, 2017
+*
+* Revisions:
+* Edited By : Name 2016/JAN/25 - Description
+*
+* Designer: Jordan Lin
+*
+* Author: Jordan Lin
+*
+* Notes:
+* Map Class functions used for reading in a map file and loading the map into the game.
+------------------------------------------------------------------------------*/
+#include "Map.h"
+
+using namespace std;
+
+/**
+ * Date: Mar. 23, 2017
+ * Author: Jordan Lin
+ * Function Interface: Map(string file)
+ * Parameters:
+ *      file - File name, must be a CSV File
+ * Description:
+ * Map Object Constructor, creates a map object using the map selected file name.
+ */
+Map::Map(string file):fname(file) {
+    memset(AIMap.data(), 0, AIMap.size() * sizeof(AIMap[0]));
+    memset(mapdata.data(), 0,  mapdata.size() * sizeof(mapdata[0]));
```

Comment:
This is awkward
Before you had the arrays as C-style raw arrays, so memsetting was the correct version
Since you've changed AIMap and mapdata to C++ std::arrays, you should do something different
Please replace with `AIMap = {};` and `mapdata = {};`
That should work, but if not, let me know